### ISSUE TITLE
Deprecate Netty3 in Druid

### ISSUE THREAD

**Issue by AnmolSun** [apache/druid#15651]
### Description

Druid still depends on Netty3 which has vulnerabilities, such as [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487). Certain downstream applications such as Hive are unable to completely remove the netty3 dependencies due to being dependent on druid. Please ref [HIVE-25013](https://issues.apache.org/jira/browse/HIVE-25013). Druid should remove its dependency on Netty3.

**Comment by dpippenger** [comment]
It would help to have some clarification from the project maintainers what the plan is for netty3 since the included 3.10.6 is no longer maintained by the netty developers and was last updated in 2016. Netty4 seems to have been added to the build several major releases ago, but it still seems that netty3 is required.

There was a PR a couple years ago that seemed to be intended to migrate off netty3 entirely, but was auto-closed
https://github.com/apache/druid/pull/14479

Is there any roadmap items to remove netty3 from druid?

**Comment by cryptoe** [comment]
Yes we wanna remove netty3 from druid. Feel free to pick this item up and the maintainers can help unblock.
